src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Metrics/Meter.cs

```diff
@@ -17,11 +17,22 @@ public class Meter : IDisposable
         private Dictionary<string, List<Instrument>> _nonObservableInstrumentsCache = new();
         internal bool Disposed { get; private set; }
 
+        internal static bool IsSupported { get; } = InitializeIsSupported();
+
+        private static bool InitializeIsSupported() =>
+            AppContext.TryGetSwitch("System.Diagnostics.Metrics.IsSupported", out bool isSupported) ? isSupported : true;
+
         /// <summary>
         /// Initialize a new instance of the Meter using the <see cref="MeterOptions" />.
         /// </summary>
         public Meter(MeterOptions options)
         {
+            if (!IsSupported)
+            {
+                Name = options is not null && options.Name is not null ? options.Name : throw new ArgumentNullException(nameof(options));
```

Comment:
It seems odd to me that we would initialize the Name property but not the other properties. Is there a reason to split it this way?

Reply:
I only added Name because the compiler was complaining not initializing non-nullable member. I am ok either way, either suppress the compiler error and not initialize the name or initialize all other properties. I don't have strong opinion which way we should go with.

Reply:
How about we initialize name to empty string? Keeps the compiler happy and preserves consistency that when metrics aren't supported none of constructor inputs are preserved.

Reply:
It might make sense to just move this check into the `Initialize` method - then we don't need to do this mess with `Name` and in 2 places.